src/entities/trade.ts

```diff
@@ -68,9 +64,9 @@ export interface BestTradeOptions {
  */
 export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType extends TradeType> {
   /**
-   * The route of the trade, i.e. which pools the trade goes through.
+   * The routes of the trade, i.e. which pools the trade goes through.
    */
-  public readonly route: Route<TInput, TOutput>
+  public readonly routes: { percent: Percent; route: Route<TInput, TOutput> }[]
```

Comment:
hmm, a little worried about `percent`. even though in theory a `Percent` can be arbitrarily precise, it's obviously possible to pass e.g 50% with an input amount of 101, which might lead to some undesirable rounding behavior

Reply:
Good call out, chatted with Moody and he proposed carrying over the remainder into the next percentage. And in total it should all net out. I implemented it in the latest revision lmk what you think